repos/spack_repo/builtin/packages/nvshmem/package.py

```diff
@@ -42,22 +52,86 @@ class Nvshmem(MakefilePackage, CudaPackage):
         when="@2.6:",
         description="Build with support for GPU initiated communication",
     )
+    variant("libfabric", default=False, description="Build with Libfabric support")
+
+    generator("ninja")
+
     conflicts("~cuda")
 
     def url_for_version(self, version):
         ver_str = "{0}".format(version)
         directory = ver_str.split("-")[0]
-        url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_{1}.txz"
+        if version < Version("3.3.9"):
+            url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_{1}.txz"
+        else:
+            url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_cuda12-all-all-{0}.tar.gz"
         return url_fmt.format(directory, version)
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+
+    depends_on("cuda@11:", when="@3.2.5:")
+
+    with default_args(when="build_system=cmake", type="build"):
+        depends_on("cmake@3.19:")
+        depends_on("ninja")
 
     depends_on("mpi", when="+mpi")
+
     depends_on("ucx", when="+ucx")
+    depends_on("ucx@1.10.0:", when="@3: +ucx")
+
     depends_on("gdrcopy", when="+gdrcopy")
+    conflicts("~gdrcopy", when="~ucx")
+    depends_on("gdrcopy@2.0:", when="@3: +gdrcopy")
+
     depends_on("nccl", when="+nccl")
+    depends_on("nccl@2.0:", when="@3: +nccl")
+
+    depends_on("libfabric", when="+libfabric")
+    depends_on("libfabric@1.15.0.0:", when="@3: +libfabric")
```

Comment:
Minor: can this be
```suggestion
    depends_on("libfabric@1.15:", when="@3: +libfabric")
```
?

Same for `nccl` and `gdrcopy` above?

Reply:
see https://github.com/spack/spack-packages/pull/504/commits/e61b744a68d74ffcafea2a0ef313f2650822e70d